### ISSUE TITLE
Is the example from the README really a round-trip?

### ISSUE THREAD

**Issue by JeffreyBenjaminBrown** [facebookincubator/retrie#11]
If your original code included some definitions of the form " map (f . g) xs", and you ran `retrie --adhoc "forall f g xs. map f (map g xs) = map (f . g) xs"` and then `retrie --adhoc "forall f g xs. map (f . g) xs = map f (map g xs)"`, it seems like your code after that supposed round-trip would no longer include any such definitions.

**Comment by xich** [comment]
I'm not sure I understand. If the original code included `map (f . g) xs`, then the running:

    retrie --adhoc "forall f g xs. map f (map g xs) = map (f . g) xs"

would do nothing, because that rule is trying to match `map f (map g xs)`.

(Running the second rule would make some changes, of course.)

**Comment by JeffreyBenjaminBrown** [comment]
Right. The first call to retrie changes a set of things we can call X. The second call to retrie undoes all the changes to X, but it introduces new changes. So calling the two in sequence does not leave the code unchanged.

Right? And if so, is there any way to undo *only* the changes from the first call?

**Comment by xich** [comment]
Ah, I see what you mean.

Yes, retrie has no notion of history. But your version control system does. Typically, we'd just selectively revert changes we don't want. (You could also run the second rule with the `--target-file` flag to only target certain files, etc.)

**Comment by JeffreyBenjaminBrown** [comment]
Cool.

Might I suggest, then, changing the line `Now you have your original code back.` in the README? Or I could submit a pull request if you'd prefer.

**Comment by xich** [comment]
What would you put there instead?

That sentence is written with the assumption that `MyModule` was your original code. On that particular example, the two rewrites are a roundtrip... it wasn't meant to be a general statement about roundtripping any code.

I'm happy to change to make it more clear though!

**Comment by JeffreyBenjaminBrown** [comment]
You could say, "Now you have undone the changes from your first call to retrie. (But note that depending on your codebase, you may have introduced new changes.)"
